src/bundles/files.js

```diff
@@ -119,26 +119,31 @@ function readAsBuffer (file) {
   })
 }
 
-bundle.doFilesWrite = (root, files) => ({dispatch, getIpfs, store}) => {
+bundle.doFilesWrite = (root, files) => async ({dispatch, getIpfs, store}) => {
   dispatch({ type: 'FILES_WRITE_STARTED' })
 
-  let promises = []
+  try {
+    let promises = []
 
-  for (const file of files) {
-    promises.push(readAsBuffer(file))
-  }
+    for (const file of files) {
+      promises.push(readAsBuffer(file))
```

Comment:
This may be a separate PR, but we may want to replace readAsBuffer with pull stream to improve performance when adding big files:
- https://www.npmjs.com/package/filereader-pull-stream
- `files2stream()`: https://github.com/ipfs-shipyard/ipfs-companion/blob/b1bf771a37fbf0d6d11fab06317a8144a8c4962b/add-on/src/popup/quick-upload.js#L29-L41

Reply:
Yes please. Is ok for separate PR, but we've gotta make this the best file uploader we can.